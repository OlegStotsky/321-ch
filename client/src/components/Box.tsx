import * as React from "react";

interface IBoxProps {
  headerMessage: string;
  children?: React.ReactNode;
}

const Box: React.SFC<IBoxProps> = ({ headerMessage, children }) => {
  return (
    <div className="box">
      <h1 className="box__header">{headerMessage}</h1>
      <div className="box__content">{children}</div>
    </div>
  );
};

export default Box;
