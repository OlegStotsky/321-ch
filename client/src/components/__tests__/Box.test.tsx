import * as React from "react";
import * as Enzyme from "enzyme";
import * as Adapter from "enzyme-adapter-react-16";
import Box from "../Box";

Enzyme.configure({ adapter: new Adapter() });

describe("TextLink", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = Enzyme.shallow(<Box headerMessage="some message" />);
  });

  it("matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("has div with box class name", () => {
    expect(wrapper.find(".box")).toHaveLength(1);
  });
  it("has h1 with class name box__header", () => {
    expect(wrapper.find("h1.box__header")).toHaveLength(1);
  });
  it("has div with class name box__content", () => {
    expect(wrapper.find("div.box__content")).toHaveLength(1);
  });
  it("provides content correctly", () => {
    wrapper = Enzyme.shallow(<Box headerMessage="123">abcd</Box>);
    expect(wrapper.contains("abcd")).toBeTruthy();
  });
});
