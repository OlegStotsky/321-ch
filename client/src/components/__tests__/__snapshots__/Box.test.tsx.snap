// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextLink matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Box
    headerMessage="some message"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="box__header"
        >
          some message
        </h1>,
        <div
          className="box__content"
        />,
      ],
      "className": "box",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "some message",
          "className": "box__header",
        },
        "ref": null,
        "rendered": "some message",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "box__content",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="box__header"
          >
            some message
          </h1>,
          <div
            className="box__content"
          />,
        ],
        "className": "box",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "some message",
            "className": "box__header",
          },
          "ref": null,
          "rendered": "some message",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "box__content",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
